CSharp/JobPrep/Program.cs

```diff
@@ -0,0 +1,333 @@
+﻿#define VERBOSE
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Batch.Auth;
+using Microsoft.Azure.Batch.Common;
+
+
+namespace JobPrep
+{
+    public class Program
+    {
+        private const string Url = "https://batch.core.windows.net";
+
+        // insert your batch account name and key along with the name of a pool to use. If the pool is new or has no VMs, 3
+        // VMs will be added to the pool to perform work.
+        private static string BatchAccount = "<batch_account>";
+        private static string BatchKey = "<batch_key>";
+        private const string PoolName = "HelloWorld-Pool";
+
+        // similarly, you need a storage account for the file staging example
+        private const string StorageAccount = "<storage_account>";
+        private const string StorageKey = "<storage_key>";
+        private const string StorageBlobEndpoint = "https://" + StorageAccount + ".blob.core.windows.net";
+
+        public static void Main(string[] args)
+        {
+            // This will boost parallel submission speed for REST APIs. If your use requires many simultaneous service calls set this number to something large, such as 100.  
+            // See: http://msdn.microsoft.com/en-us/library/system.net.servicepointmanager.defaultconnectionlimit%28v=vs.110%29.aspx for more info.
+            System.Net.ServicePointManager.DefaultConnectionLimit = 20;
+
+            if (File.Exists("..\\..\\batchcred.txt"))
+            {
+                string[] lines = File.ReadAllLines("..\\..\\batchcred.txt");
+                if (lines.Length >= 2)
+                {
+                    BatchAccount = lines[0];
+                    BatchKey = lines[1];
+                }
+            }
+
+            // Get an instance of the BatchClient for a given Azure Batch account.
+            BatchCredentials cred = new BatchCredentials(BatchAccount, BatchKey);
+            using (IBatchClient client = BatchClient.Connect(Url, cred))
+            {
+                // if you want to put a retry policy in place, enable it here
+                // the built-in policies are No Retry (default), Linear Retry, and Exponential Retry
+                //client.CustomBehaviors.Add(new SetRetryPolicy(new Microsoft.Azure.Batch.Protocol.LinearRetry()));
+
+                ListPools(client);
+                ListWorkItems(client);
+
+                CreatePoolIfNotExist(client, Program.PoolName);
+                AddWork(client);
+
+                ListPools(client);
+                ListWorkItems(client);
+            }
+
+            Console.WriteLine("Press return to exit...");
```

Comment:
I envision this kind of common stuff going into a common assembly. This would include cred mgtm and pool retention policy. Rather than reimplement 10 different ways across the various samples